Cleanup scoring (#466)
Kattis/problem-package-format: spec/2023-07-draft.md

Each test group may define a static validation test case.
It is an error to define static validation test cases without providing a static validator.
A static validation test case is defined within a group's `test_group.yaml` file by specifying the key `static_validation_score`.

> If `static_validation_score` is specified as an non-negaive integer is it the maximum score of the static validation test case (see [Scoring Problems](#scoring-problems) for details).

Reviewer: ```suggestion
If `static_validation_score` is specified as a non-negative integer, then it is the maximum score of the static validation test case (see [Scoring Problems](#scoring-problems) for details).
```

> If it is specified as `pass-fail` then the test group it is part of must have `pass-fail` aggregation, or the problem be of type `pas-fail`.

Reviewer: ```suggestion
If it is specified as `pass-fail`, then the test group it is part of must have `pass-fail` aggregation, or the problem must be of type `pass-fail`.
```

> If `static_validator_args` is given it defines arguments passed to the static validator.

Reviewer: ```suggestion
If `static_validator_args` is given, then it defines arguments passed to the static validator.
```

It is an error to:
- provide a static validator for `submit-answer` type problems,
- specify a `static_validation_score` in a test group with `pass-fail` aggregation or a problem that does not have the type `scoring`,

> - specify a `static_validator_args` without specifying `static_validation_score`.

Reviewer: ```suggestion
- specify `static_validator_args` without specifying `static_validation_score`.
```
